06_functions/main.go

```diff
@@ -123,13 +123,13 @@ func printInput(str string, times int) {
 // Call the function a couple of times with different values and print the
 // results.
 func concatString(str string, times int) string {
-	var count = 1
 	for {
-		if count > times {
+		if times < 1 {
 			break
 		}
 		str += str
-		count++
+		times--
+		fmt.Println(times)
 	}
 	return str
```

Comment:
@nobe4 Tried your suggestion. But now it causes to string to increase too much.
You are certain about `< 0` right? I will give it a try later again.

Reply:
maybe `< 0` is too much, but the idea is to decrement `times` as you found out :)